client/verta/tests/test_versioning/test_dataset.py

```diff
@@ -165,6 +170,72 @@ def test_repr(self):
 
         assert dataset_ver.__repr__()
 
+    def test_mngd_ver_file(self, commit, in_tempdir):
+        filename = "tiny1.bin"
+        s3_key = "s3://verta-versioned-bucket/tiny-files/{}".format(filename)
+        blob_path = "data"
+
+        dataset = verta.dataset.S3(s3_key, enable_mdb_versioning=True)
+        commit.update(blob_path, dataset)
+        commit.save("Version data.")
+        dataset = commit.get(blob_path)
+
+        # download to implicit path
+        filepath = dataset.download(s3_key)
+        assert os.path.isfile(filepath)
+        assert filepath == os.path.abspath(filename)
+
+        # download to implicit path without collision
+        filepath2 = dataset.download(s3_key)
+        assert os.path.isfile(filepath2)
+        assert filepath2 != filepath
+
+        # download to explicit path with overwrite
+        last_updated = os.path.getmtime(filepath)
+        filepath3 = dataset.download(s3_key, filepath)
+        assert filepath3 == filepath
+        assert os.path.getmtime(filepath) > last_updated
+
+    def test_mngd_ver_folder(self, commit, in_tempdir):
+        dirname = "tiny-files/"
+        s3_folder = "s3://verta-versioned-bucket/{}".format(dirname)
+        blob_path = "data"
+
+        dataset = verta.dataset.S3(s3_folder, enable_mdb_versioning=True)
+        commit.update(blob_path, dataset)
+        commit.save("Version data.")
+        dataset = commit.get(blob_path)
+
+        # download to implicit path
+        dirpath = dataset.download(s3_folder)
+        assert os.path.isdir(dirpath)
+        assert dirpath == os.path.abspath(dirname)
+
+        # download to implicit path without collision
+        dirpath2 = dataset.download(s3_folder)
+        assert os.path.isdir(dirpath2)
+        assert dirpath2 != dirpath
+
+        # download to explicit path with overwrite
+        last_updated = os.path.getmtime(dirpath)
+        dirpath3 = dataset.download(s3_folder, dirpath)
+        assert dirpath3 == dirpath
+        assert os.path.getmtime(dirpath) > last_updated
+
+    def test_not_to_s3_dir(self, commit, in_tempdir):
+        """If the user specifies "s3://", things shouldn't go into a """
+        dirname = "tiny-files/"
+        s3_folder = "s3://verta-versioned-bucket/{}".format(dirname)
+        blob_path = "data"
+
+        dataset = verta.dataset.S3(s3_folder, enable_mdb_versioning=True)
+        commit.update(blob_path, dataset)
+        commit.save("Version data.")
+        dataset = commit.get(blob_path)
+
+        dirpath = dataset.download("s3://")
```

Comment:
What happens when it's empty?

Reply:
If the string is empty? The user gets a
```
KeyError: 'no components found for path '
```

See my comment above about download-all behavior

Reply:
The "comment above" disappeared.
>I have VR-4875 for dataset.download() with no args to mean "download all"